third_party/move/move-vm/runtime/src/storage/struct_name_index_map.rs

```diff
@@ -1,43 +1,96 @@
 // Copyright (c) The Move Contributors
 // SPDX-License-Identifier: Apache-2.0
 
+use claims::assert_none;
+use move_core_types::language_storage::{StructTag, TypeTag};
 use move_vm_types::loaded_data::runtime_types::{StructIdentifier, StructNameIndex};
 use parking_lot::RwLock;
-use std::collections::BTreeMap;
+use std::{collections::BTreeMap, sync::Arc};
 
 #[derive(Clone)]
 struct IndexMap<T: Clone + Ord> {
     forward_map: BTreeMap<T, usize>,
-    backward_map: Vec<T>,
+    backward_map: Vec<Arc<T>>,
```

Comment:
I wonder if we should get in the habit of using https://docs.rs/triomphe/latest/triomphe/ when we don't need weak pointers? @ziaptos @runtian-zhou 

In the future maybe we won't even need Arc's in a lot of places where ownership is clear.